packages/opencensus-core/src/exporters/console-exporter.ts

```diff
@@ -33,8 +33,8 @@ export class NoopExporter implements types.Exporter {
 /** Format and sends span data to the console. */
 export class ConsoleExporter implements types.Exporter {
   /** Buffer object to store the spans. */
-  private buffer: ExporterBuffer;
   private logger: loggerTypes.Logger;
+  buffer: ExporterBuffer;
```

Comment:
Please keep this private.

Reply:
I had to make it public so I could access the buffer queue from the tests, also, I think the exporter buffer should be accessible outside the exporter. But we can accomplish this in different ways, if you prefer, I could set it private and make a get method or return the buffer queue at `onEndSpan()` (this last one is not my favorite). We could also think about other ways or if accessing the exporter buffer is really necessary.

Reply:
I am typically against making class members public for testing, because they are implementation details, and we end up testing behavior that is not observable by normal users of the class. The same goes for getters for class members.

There are a few ways to work around this:
1) We can test _only_ observable behavior: initialize the `ConsoleExporter` with a buffer of size `X`, add `X-1` spans, verify that nothing has been written to stdout yet, then add one more and verify that all spans were written to stdout.
2) We can use dependency injection: Instead of initializing `this.buffer` in the constructor with a `new ExporterBuffer`, pass it as a parameter and have the caller create the `ExporterBuffer` object. This way, you don't have to test the behavior of `ConsoleExporter` that relies on `ExporterBuffer`: for example, the behavior in (1) doesn't really pertain to `ConsoleExporter` at all, it is a property of `ExporterBuffer`. Instead, the behavior you are testing is that `ConsoleExporter` correctly uses `ExporterBuffer` -- and this is possible by just passing in a mock `ExporterBuffer` implementation.

I prefer (2) and think it is a more longer-term solution.

Reply:
I understand your point. I really like the dependency injection approach but to create a buffer I need an exporter. I could change the `ExporterBuffer` constructor to accept an exporter optionally and set it latter, but that simple change would affect many exporters, if not all. So, I ended up doing your first suggestion, I intercepted stdout before and after the `bufferSize` threshold.